libs/blocks/mobile-app-banner/mobile-app-banner.js

```diff
@@ -59,7 +59,7 @@ function branchInit(key) {
 /* eslint-enable */
 export default async function init(el) {
   const header = document.querySelector('.global-navigation');
-  if (!header) return;
+  if (!header || header.classList.contains('has-promo')) return;
```

Comment:
```suggestion
if (!header?.classList.contains('has-promo')) return;
```
if header is not available, the execution will stop

Reply:
If header is not available we want to stop the execution.
```if (!header?.classList.contains('has-promo')) return;```
this condition fail in the case if we have `has-promo` class available.
So basically we are checking 
if header is not available then return
if header is available the check for `has-promo` class if it is present then again return.

Reply:
you are correct, my bad